p2p/peer.go

```diff
@@ -492,6 +492,34 @@ func (m *PeerManager) Add(address PeerAddress) error {
 	return nil
 }
 
+// Advertise returns a list of peer addresses to advertise to a peer.
+//
+// FIXME: We currently just pass all addresses we have, which is very naïve. We
+// should e.g. only send addresses that the peer can actually reach (by
+// resolving the addresses into endpoints and making sure any private IP
+// addresses are on the same network as the remote peer endpoint). However, this
+// would require resolving endpoints either here (too slow) or generally in the
```

Comment:
would it make sense to only resolve the first endpoint? if not successful, we continue resolving 2nd, 3rd ... N endpoints

Reply:
Hm? Resolving an address will always return all endpoints, that's how DNS works. The problem is that we may want to advertise e.g. 100 addresses, and resolving all 100 addresses into endpoints on demand would be slow. We may want to have the peer manager resolve the endpoints when it's first given the address and then periodically refresh it.

Reply:
> We may want to have the peer manager resolve the endpoints when it's first given the address and then periodically refresh it.

that sounds fine to me